scintilla/boostregex/BoostRegExSearch.cxx

```diff
@@ -352,7 +353,7 @@ BoostRegexSearch::Match BoostRegexSearch::EncodingDependent<CharT, CharacterIter
 			? search._boostRegexFlags & ~regex_constants::match_not_bol
 			: search._boostRegexFlags |  regex_constants::match_not_bol;
 		const bool end_reached = next_search_from_position > search._endPosition;
-		found = !end_reached && boost::regex_search(CharacterIterator(search._document, next_search_from_position, search._endPosition), endIterator, _match, _regex, search._boostRegexFlags);
+		found = !end_reached && boost::regex_search(CharacterIterator(search._document, next_search_from_position, search._endPosition), endIterator, _match, _regex, search._boostRegexFlags, baseIterator);

```

Comment:
This is interesting.  I don't see this form of the call to `regex_search` in this documentation?: 
https://www.boost.org/doc/libs/1_74_0/libs/regex/doc/html/boost_regex/ref/regex_search.html
What am I missing here?

Reply:
In the documentation, you missed nothing, as far as I see. It seems to be outdated or incomplete.

I came to the solution by examining the source code.